cli/internal/client/client.go

```diff
@@ -301,7 +301,7 @@ func (c *ApiClient) PutArtifact(hash string, artifactBody []byte, duration int,
 
 // FetchArtifact attempts to retrieve the build artifact with the given hash from the
 // Remote Caching server
-func (c *ApiClient) FetchArtifact(hash string) (*http.Response, error) {
+func (c *ApiClient) FetchArtifact(hash string, assertOnly bool) (*http.Response, error) {
```

Comment:
I think rather than expose a boolean parameter here, we would be better off with a separate, top-level API. Something like `ArtifactExists` or `HeadArtifact` or something along those lines. You can funnel it down to the same function internal to the client (maybe pass in the method or something?), but that will make it clearer to users vs having to understand what `true` or `false` means in the context of `FetchArtifact`.